pallets/foreign-investments/src/tests.rs

```diff
@@ -1426,30 +1600,23 @@ mod redemption {
 
 			util::process_redemption(3 * TRANCHE_AMOUNT / 4);
 
-			assert_ok!(ForeignInvestment::collect_foreign_redemption(
-				&USER,
-				INVESTMENT_ID,
-				FOREIGN_CURR
-			));
+			assert_ok!(MockInvestment::collect_redemption(USER, INVESTMENT_ID));
 
 			util::fulfill_last_swap(Action::Redemption, tranche_to_pool(TRANCHE_AMOUNT / 2));
 
-			MockCollectRedeemHook::mock_notify_status_change(|(who, investment_id), msg| {
-				assert_eq!(who, USER);
-				assert_eq!(investment_id, INVESTMENT_ID);
-				assert_eq!(
-					msg,
-					ExecutedForeignCollect {
-						currency: FOREIGN_CURR,
-						amount_currency_payout: pool_to_foreign(tranche_to_pool(
-							3 * TRANCHE_AMOUNT / 4
-						)),
-						amount_tranche_tokens_payout: 3 * TRANCHE_AMOUNT / 4,
-						amount_remaining: TRANCHE_AMOUNT / 4,
-					}
-				);
-				Ok(())
-			});
+			MockHooks::mock_fulfill_collect_redemption(
```

Comment:
Isn't this mock ineffective because we don't collect anymore?

Reply:
No, because we only collect when the entire swap is swapped, and this happens in the next line.

Note that who performs the `fulfill_collect_redemption is the last partial swap from the collected amount.